storage/rocksdb/ha_rocksdb.cc

```diff
@@ -4331,8 +4356,26 @@ void ha_rocksdb::load_auto_incr_value() {
   const bool save_keyread_only = m_keyread_only;
   m_keyread_only = true;
 
-  if (!index_last(table->record[0]))
-    update_auto_incr_val();
+  auto found_key = m_tbl_def->get_pk();
+
+  auto scf = dict_manager.get_system_cf();
+
+  // Construct the key of the data dictionary to get the auto_increment value
+  uchar key_buf[Rdb_key_def::INDEX_NUMBER_SIZE * 3] = {0};
+  dict_manager.get_key_from_index_id((uchar *)key_buf,
```

Comment:
I think this code for fetching and writing the contents of the auto_incr key should reside within dict_manager. The dict_manager is meant to handle putting/deleting keys to the system column family. The merge operations should go through dict_manager instead of being pulled out here.

Reply:
the implementation of the Rdb_transaction class is defined in ha_rocksdb.cc and Rdb_dict_manager is unable to access the Rdb_transaction's merge() and get(), so the best solution would be to somehow extract the key from the Rdb_dict_manager and call merge() or get() in ha_rocksdb. This would either require some wrapper around dump_index_id or extension of dump_index_id to the public section.

Reply:
Rdb_binlog_manager does something similar with the system column family for each Rdb_transaction during commit(). It pulls out the writebatch of the Rdb_transaction class and passes it as an argument to Rdb_binlog_manager. The binlog_manager then writes the key/value via writebatch->put(). The writebatch also has a merge() interface. My original thought was to record the highest auto increment value generated for each gl_index_id/table and store it local to Rdb_transaction. On commit, these merge keys can be added to the write batch.

If you decide to go with pulling out the key to ha_rocksdb.cc for reading and merging, the functions should be consolidated into common routines that the different call sites invoke. This routine reading the key should also perform the version checks, etc.